### PR TITLE
Enhance algorithms explanation

### PR DESCRIPTION
**Pull request by rbrox**
### Description
- Added a recurrence relation to provide deeper insight into the time complexity analysis.
- Addressed minor alignment issues for improved readability.

Files/Paths Affected:
- en/Sorting Algorithms/Merge Sort.md

### Checklist

- [Y] A description of the changes in this PR is mentioned above.
- [Y] All the new and existing checks pass.
- [Y] No plagiarized, duplicated, or repetitive documentation that has been directly copied from another source.
- [Y] If it's a new explanation, it contains solid, understandable, and accessible information.
- [Y] I have read the whole [contributing guidelines](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/CONTRIBUTING.md) and agree to the [Code of Conduct](https://github.com/TheAlgorithms/.github/blob/master/CODE_OF_CONDUCT.md).

|      Original       |  

image
|     Updated       |
image

### CODE REVIEW

File: en/Sorting Algorithms/Merge Sort.md [hunk 1]
```diff
@@ -18,9 +18,17 @@ Average - O(n log n)
...
+
+$$T(n) = 2T\left(\frac{n}{2}\right) + An$$
+
+
+
 #### Space Complexity
 
-`O(n)`
+```
+O(n)
```
Comment (appgurueu) [inline en/Sorting Algorithms/Merge Sort.md:30]: Should be `$$O(n)$$`, IMO. Also consider adding an explanation.

File: en/Sorting Algorithms/Merge Sort.md [hunk 2]
```diff
@@ -18,9 +18,17 @@ Average - O(n log n)
 Worst case - O(n log n)
 ```
 
+The time complexity can be obtained by solving the below recurrence relation:
+
+$$T(n) = 2T\left(\frac{n}{2}\right) + An$$
```
Comment (appgurueu) [inline en/Sorting Algorithms/Merge Sort.md:23]: Is `A` defined anywhere? Is it just an arbitrary constant? This is unclear currently.

Comment (rbrox) [reply en/Sorting Algorithms/Merge Sort.md:23]: So should I define `A` like solve the whole equation? Would that be appreciated?

File: en/Sorting Algorithms/Merge Sort.md (hunk 1 above)
Comment (rbrox) [reply en/Sorting Algorithms/Merge Sort.md:30]: I changed it to ''' O(n) ''', for consistent display, the time complexity displays in a code block, whereas the space complexity displays as an inline code segment.

Comment (rbrox) [reply en/Sorting Algorithms/Merge Sort.md:30]: I'll continue with whatever you discern to be better though.

File: en/Sorting Algorithms/Merge Sort.md (hunk 2 above)
Comment (rbrox) [reply en/Sorting Algorithms/Merge Sort.md:23]: I understand. Will update with a proper derivation also define all used constants

File: en/Sorting Algorithms/Merge Sort.md (hunk 1 above)
Comment (appgurueu) [reply en/Sorting Algorithms/Merge Sort.md:30]: I think `$...$` (inline formula) or `$$...$$` is more appropriate for typesetting math. The (ab)use of code blocks and inline code was mostly a workaround for platforms that lack(ed) support for LaTeX, I think.

**Pull request closed without merging**